src/scores/utils.py

```diff
@@ -53,87 +53,7 @@ class DimensionError(Exception):
     """
 
 
-def gather_dimensions(  # pylint: disable=too-many-branches
-    fcst_dims: Iterable[Hashable],
-    obs_dims: Iterable[Hashable],
-    reduce_dims: FlexibleDimensionTypes = None,
-    preserve_dims: FlexibleDimensionTypes = None,
-) -> set[Hashable]:
-    """
-    Establish which dimensions to reduce when calculating errors but before taking means.
-
-    Note: `scores.utils.gather_dimensions` and `scores.utils.gather_dimensions2` will be
-    integrated at some point in the future. `scores.utils.gather_dimensions2` offers
-    more comprehensive and less restrictive dimension checking and should be preferred in
-    the meantime. See `scores.probability.crps_cdf` for an example of
-    `scores.utils.gather_dimensions2` usage.
-
-    Args:
-        fcst_dims: Forecast dimensions inputs
-        obs_dims: Observation dimensions inputs.
-        reduce_dims: Dimensions to reduce.
-        preserve_dims: Dimensions to preserve.
-
-    Returns:
-        Dimensions based on optional args.
-    Raises:
-        ValueError: When `preserve_dims and `reduce_dims` are both specified.
-
-    See also:
-        `scores.utils.gather_dimensions2`
-    """
-
-    all_dims = set(fcst_dims).union(set(obs_dims))
-
-    # Handle error conditions related to specified dimensions
-    if preserve_dims is not None and reduce_dims is not None:
-        raise ValueError(ERROR_OVERSPECIFIED_PRESERVE_REDUCE)
-
-    # Handle error conditions related to specified dimensions
-    specified = preserve_dims or reduce_dims
-    if specified == "all":
-        if "all" in all_dims:
-            warnings.warn(WARN_ALL_DATA_CONFLICT_MSG)
-    elif specified is not None:
-        if isinstance(specified, str):
-            specified = [specified]
-
-        if not set(specified).issubset(all_dims):
-            if preserve_dims is not None:
-                raise ValueError(ERROR_SPECIFIED_NONPRESENT_PRESERVE_DIMENSION)
-            raise ValueError(ERROR_SPECIFIED_NONPRESENT_REDUCE_DIMENSION)
-
-    # Handle preserve_dims case
-    if preserve_dims is not None:
-        if preserve_dims == "all":
-            return set([])
-
-        if isinstance(preserve_dims, str):
-            preserve_dims = [preserve_dims]
-
-        reduce_dims = set(all_dims).difference(preserve_dims)
-
-    # Handle reduce all
-    elif reduce_dims == "all":
-        reduce_dims = set(all_dims)
-
-    # Handle is reduce_dims and preserve_dims are both None
-    if reduce_dims is None and preserve_dims is None:
-        reduce_dims = set(all_dims)
-
-    # Handle reduce by string
-    elif isinstance(reduce_dims, str):
-        reduce_dims = set([reduce_dims])
-
-    # Turn into a set if needed
-    assert reduce_dims is not None
-    reduce_dims = set(reduce_dims)
-
-    # Reduce by list is the default so no handling needed
-    return reduce_dims
-
-
-def gather_dimensions2(
+def gather_dimensions(
```

Comment:
The implementation logic of the new combined logic looks good to me after careful reading.